### DISCUSSION TITLE
How could a caller use a task without loading the library that task uses?

### DISCUSSION THREAD

**Discussion by ylchan87** [celery/celery#9115] [Q&A]
From the doc, to call a task we need to import the py file where the task is defined:
https://docs.celeryq.dev/en/stable/getting-started/first-steps-with-celery.html#calling-the-task

i.e.  task.py
```
from celery import Celery

app = Celery('tasks', broker='pyamqp://guest@localhost//')

@app.task
def add(x, y):
    return x + y
```

and to call the `add` task, task.py is imported
```
from tasks import add
add.delay(4, 4)
```

Now, what if the add task uses some GPU library, and could not be installed on the caller machine, which has no GPU at all?

Is there a way I could expose only the call signature of the task, and hide all implementation details?

Not sure if this is a python programming question or a celery one, but any hint would be much appreciated.

Thanks!

**Comment by thedrow** [comment]
Use [`send_task`](https://docs.celeryq.dev/en/stable/reference/celery.html#celery.Celery.send_task) to send a task that is unregistered in your app.
